### ISSUE TITLE
Model Architecture Bug: Share with user added data broken

### ISSUE THREAD

**Issue by soyarsauce** [TerriaJS/terriajs#3602]
Steps to reproduce
1. Hit http://ci.terria.io/mobx/
2. Click "Add data" in the workbench
3. My Data tab -> add any data (local + web has same effect) (`http://ereeftds.bom.gov.au/ereefs/tds/wms/ereefs/mwq_gridAgg_P1A` for a test case)
4. Close explorer window (Click done)
5. Click Share/Print button
Observe white page

```
SharePanel.jsx:305 Uncaught TypeError: Cannot read property 'map' of undefined
    at Object.renderWarning (SharePanel.jsx:305)
    at Object.renderContentWithPrintAndEmbed (SharePanel.jsx:360)
    at Object.renderContent (SharePanel.jsx:324)
    at Object.SharePanel_render [as render] (SharePanel.jsx:493)
    at ObserveModelMixin.js:34
    at Function.xd (knockout-3.5.0.js:55)
    at Function.yd (knockout-3.5.0.js:55)
    at Function.ha (knockout-3.5.0.js:54)
    at Object.a.o.a.$ (knockout-3.5.0.js:52)
    at ObserveModelMixin.js:25
```

**Comment by soyarsauce** [comment]
No longer crashes now but user added data doesn't get shared

**Comment by soyarsauce** [comment]
Almost wrote another issue for this, but in addition to user share not working atm, if you add normal catalog items to workbench as well as have user added data in your session, creating a share link will mean not being able to share the normal catalog items as well.

**Comment by tephenavies** [comment]
Fixed in #3929.
